### ISSUE TITLE
Wiremock continues to perform checks after one of the checks is negative

### ISSUE THREAD

**Issue by Daemon2017** [wiremock/wiremock#1611]
Hello! 

I'm using Wiremock embedded (org.springframework.cloud:spring-cloud-contract-wiremock:2.2.5 with wiremock-2.21.0 inside it) for testing my Spring-application: it have 3 stubs for 3 third party services - 1 SOAP, 1 REST and 1 web-form. 

The problem is too much logs of INFO-level: 
1) When my Spring-application calls web-form stub, I see 102 records in logs -
> 2021-10-01 13:13:00[qtp683039119-39]INFO  WireMock - Warning: failed to parse the XML document
> XML: type=creds&blablabla

and then 

> 2021-10-01 13:13:00[qtp683039119-39]INFO  WireMock - Request received:
2) When my Spring-application calls REST stub, I see 102 records in logs -
> 2021-10-01 13:13:00[qtp683039119-33]INFO  WireMock - Warning: failed to parse the XML document
> XML:

and then

> 2021-10-01 13:13:00[qtp683039119-33]INFO  WireMock - Request received:
3) When my Spring-application calls SOAP stub, I'm getting 
> 2021-10-01 13:53:36[qtp854444123-33]INFO  WireMock - Request received:

I've tried to find the reason for the strange number 102 and then I've realised that I have 102 checks in my stub for SOAP service!
In my opinion, it's a bug: **Wiremock should stop all further checks in the stub if 1 check in this stub already failed, isn't it?**

Here is my BaseTest.java class:
```
@SpringBootTest(
        classes = {App.class},
        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
)
@AutoConfigureWireMock(port = 0, stubs = "file:src/test/resources/mocks")
public abstract class BaseTest extends AbstractTestNGSpringContextTests {
      @LocalServerPort
    protected int localPort;

    @Configuration
    protected static class Config {
        @Bean
        WireMockConfigurationCustomizer optionsCustomizer() {
            return wireMockConfiguration -> wireMockConfiguration
                    .extensions(new Webhooks())
                    .notifier(new Slf4jNotifier(true));
        }
    }
}

```
Web-form stub looks like:
```
    "bodyPatterns": [
      {
        "contains": "type=creds"
      }
// and 2 more checks
    ]
```
REST stub:
   ```
 "bodyPatterns": [
      {
        "equalToJson": "{\"abra\":\"cadabra\"}",
        "ignoreArrayOrder": true,
        "ignoreExtraElements": true
      }
    ]
```
SOAP stub:
```
"bodyPatterns": [
      {
        "matchesXPath": "//HeaderInfo[string-length(@RqUID) > 0]",
      },
     //and 101 more checks
    ]
```

**Comment by tomakehurst** [comment]
I wouldn't describe it as a bug exactly, since I assume you're still getting the correct result. Unless that's not the case?

Definitely not optimal from a performance PoV though.

I'm curious as to why you need 102 individual xpath matchers per stub, that seems like an awful lot. I generally advise keeping it to the minimum number necessary to uniquely identify the request and route to the right stub.

**Comment by Daemon2017** [comment]
Thank you for your reply!
Yes, I'm still getting the correct result.

Here I've described a perfect model situation. In real life, I have hundreds of tests (and thousands of "failed to parse" warnings) - that's why I need so many checks.

Is there a way to save useful wiremock log records (request received, request matched the stub, etc), but exclude noise log records (failed to parse, etc)?

**Comment by tomakehurst** [comment]
@Daemon2017 yes, probably the best way to do this is write a `PostServeAction` implementation that logs exactly the data you want in the form you want, then register it as an extension.

BTW there's a major log redesign coming and this kind of issue is partly what it's intended to address: https://github.com/wiremock/wiremock/issues/1471

**Comment by Daemon2017** [comment]
@tomakehurst Thank you! I hope that log redesign in future will make my task easier :)

**Comment by htmldoug** [comment]
Similar to #1153.
